[dialog] Flaky test related to resize listener when running locally

### Description

The following IT fails for me because of fractional values:

```
[ERROR]   DialogTestPageIT.resizableDialogListenerIsCalled:355 
expected:<...called with width (2[50px) and height (250]px)> but was:<...called with width (2[49.891px) and height (249.891]px)>
```

### Expected outcome

The test should pass.

### Minimal reproducible example

See `DialogTestPageIT`.

### Steps to reproduce

Run the IT and see that it fails.

```
mvn verify -Dvaadin.pnpm.enable -q -am -B -pl vaadin-dialog-flow-parent/vaadin-dialog-flow-integration-tests -Dtest=none -Dit.test=DialogTestPageIT
```

### Environment

Vaadin version(s): v24 (master)

### Browsers

Chrome

--- a/vaadin-dialog-flow-parent/vaadin-dialog-flow-integration-tests/src/main/java/com/vaadin/flow/component/dialog/tests/DialogTestPage.java
+++ b/vaadin-dialog-flow-parent/vaadin-dialog-flow-integration-tests/src/main/java/com/vaadin/flow/component/dialog/tests/DialogTestPage.java
@@ -180,6 +180,10 @@ public class DialogTestPage extends Div {
         add(button);
     }
 
+    private int getDimension(String dimension) {
+        return (int) Math.round(Float.parseFloat(dimension.replace("px", "")));
+    }
+
     private void createResizableDraggableDialog() {
         Dialog dialog = new Dialog();
         dialog.setId("dialog-resizable-draggable");
@@ -191,13 +195,14 @@ public class DialogTestPage extends Div {
         Div message = new Div();
         message.setId("dialog-resizable-draggable-message");
 
-        dialog.addResizeListener(e -> message
-                .setText("Rezise listener called with width (" + e.getWidth()
-                        + ") and height (" + e.getHeight() + ")"));
+        dialog.addResizeListener(e -> message.setText(String.format(
+                "Resize listener called with width (%d) and height (%d)",
+                getDimension(e.getWidth()), getDimension(e.getHeight()))));
 
-        dialog.addOpenedChangeListener(e -> message
-                .setText("Initial size with width (" + dialog.getWidth()
-                        + ") and height (" + dialog.getHeight() + ")"));
+        dialog.addOpenedChangeListener(e -> message.setText(
+                String.format("Initial size with width (%d) and height (%d)",
+                        getDimension(dialog.getWidth()),
+                        getDimension(dialog.getHeight()))));
 
         NativeButton closeButton = new NativeButton(CLOSE_CAPTION,
                 e -> dialog.close());

--- a/vaadin-dialog-flow-parent/vaadin-dialog-flow-integration-tests/src/test/java/com/vaadin/flow/component/dialog/tests/DialogTestPageIT.java
+++ b/vaadin-dialog-flow-parent/vaadin-dialog-flow-integration-tests/src/test/java/com/vaadin/flow/component/dialog/tests/DialogTestPageIT.java
@@ -344,8 +344,7 @@ public class DialogTestPageIT extends AbstractComponentIT {
         WebElement message = findElement(
                 By.id("dialog-resizable-draggable-message"));
 
-        Assert.assertEquals(
-                "Initial size with width (200px) and height (200px)",
+        Assert.assertEquals("Initial size with width (200) and height (200)",
                 message.getText());
 
         TestBenchElement overlayContent = getOverlayContent();
@@ -353,7 +352,7 @@ public class DialogTestPageIT extends AbstractComponentIT {
         resizeDialog(overlayContent, 50, 50);
 
         Assert.assertEquals(
-                "Rezise listener called with width (250px) and height (250px)",
+                "Resize listener called with width (250) and height (250)",
                 message.getText());
     }
 

